DigitalLearningSolutions.Web.Tests/Controllers/TrackingSystem/Delegates/AddDelegateToGroupControllerTests.cs

```diff
@@ -0,0 +1,250 @@
+                .WithMockServices()
+                .WithMockTempData();
+        }
+
+        [Test]
+        public void Index_with_no_query_parameters_uses_cookie_value_for_filterBy()
+        {
+            // When
+            var result = addDelegateToGroupsController.Index();
+
+            // Then
+            result.As<ViewResult>().Model.As<AddDelegateToGroupViewModel>().FilterBy.Should()
+                .Be("ActiveStatus|Active|false");
+        }
+
+        [Test]
+        public void Index_with_query_parameters_uses_query_parameter_value_for_filterBy()
+        {
+            // Given
+            const string filterBy = "PasswordStatus|IsPasswordSet|true";
+            A.CallTo(() => httpRequest.Query.ContainsKey("filterBy")).Returns(true);
+
+            // When
+            var result = addDelegateToGroupsController.Index(filterBy: filterBy);
+
+            // Then
+            result.As<ViewResult>().Model.As<AllDelegatesViewModel>().FilterBy.Should()
+                .Be(filterBy);
+        }
+
+        [Test]
+        public void Index_with_CLEAR_filterBy_query_parameter_removes_cookie()
+        {
+            // Given
+            const string filterBy = "CLEAR";
+
+            // When
+            var result = addDelegateToGroupsController.Index(filterBy: filterBy);
+
+            // Then
+            A.CallTo(() => httpResponse.Cookies.Delete("AddDelegateToGroupFilter")).MustHaveHappened();
+            result.As<ViewResult>().Model.As<AllDelegatesViewModel>().FilterBy.Should()
+                .BeNull();
+        }
+
+        [Test]
+        public void Index_with_null_filterBy_and_new_filter_query_parameter_adds_new_cookie_value()
+        {
+            // Given
+            const string? filterBy = null;
+            const string newFilterValue = "PasswordStatus|IsPasswordSet|true";
+
+            // When
+            var result = addDelegateToGroupsController.Index(filterBy: filterBy, filterValue: newFilterValue);
+
+            // Then
+            A.CallTo(() => httpResponse.Cookies.Append("AddDelegateToGroupFilter", newFilterValue, A<CookieOptions>._))
+                .MustHaveHappened();
+            result.As<ViewResult>().Model.As<AllDelegatesViewModel>().FilterBy.Should()
+                .Be(newFilterValue);
+        }
+
+        [Test]
+        public void Index_with_CLEAR_filterBy_and_new_filter_query_parameter_sets_new_cookie_value()
+        {
+            // Given
+            const string filterBy = "CLEAR";
+            const string newFilterValue = "PasswordStatus|IsPasswordSet|true";
+
+            // When
+            var result = addDelegateToGroupsController.Index(filterBy: filterBy, filterValue: newFilterValue);
+
+            // Then
+            A.CallTo(() => httpResponse.Cookies.Append("AddDelegateToGroupFilter", newFilterValue, A<CookieOptions>._))
+                .MustHaveHappened();
+            result.As<ViewResult>().Model.As<AllDelegatesViewModel>().FilterBy.Should()

```

Comment:
Is that the correct view model?

Reply:
I have changed this to correct view model